launcher/ui/themes/ThemeManager.cpp

```diff
@@ -59,53 +59,107 @@ ITheme* ThemeManager::getTheme(QString themeId)
-                // Load pure QSS Themes
-                QDirIterator stylesheetFileIterator(dir.absoluteFilePath(""), { "*.qss", "*.css" }, QDir::Files);
-                while (stylesheetFileIterator.hasNext()) {
-                    QFile customThemeFile(stylesheetFileIterator.next());
-                    QFileInfo customThemeFileInfo(customThemeFile);
-                    themeDebugLog() << "Loading QSS Theme from:" << customThemeFileInfo.absoluteFilePath();
-                    addTheme(std::make_unique<CustomTheme>(getTheme(darkThemeId), customThemeFileInfo, false));
-                }
+    initializeWidgets();
+}
+
+void ThemeManager::initializeIcons()
+{
+    // TODO: icon themes and instance icons do not mesh well together. Rearrange and fix discrepancies!
+    // set icon theme search path!
+    themeDebugLog() << "<> Initializing Icon Themes";
+
+    auto searchPaths = QIcon::themeSearchPaths();
+    searchPaths.append(m_iconThemeFolder.path());
+    QIcon::setThemeSearchPaths(searchPaths);
+
+    for (const QString& id : builtinIcons) {
+        IconTheme theme(id, QString(":/icons/%1").arg(id));
+        if (!theme.load()) {
+            themeWarningLog() << "Couldn't load built-in icon theme" << id;
+            continue;
+        }
+
+        addIconTheme(std::move(theme));
+        themeDebugLog() << "Loaded Built-In Icon Theme" << id;
+    }
+
+    if (!m_iconThemeFolder.mkpath("."))
+        themeWarningLog() << "Couldn't create icon theme folder";
+    themeDebugLog() << "Icon Theme Folder Path: " << m_iconThemeFolder.absolutePath();
+
+    QDirIterator directoryIterator(m_iconThemeFolder.path(), QDir::Dirs | QDir::NoDotAndDotDot);
+    while (directoryIterator.hasNext()) {
+        QDir dir(directoryIterator.next());
+        IconTheme theme(dir.dirName(), dir.path());
+        if (!theme.load())
+            continue;
+
+        addIconTheme(std::move(theme));
+        themeDebugLog() << "Loaded Custom Icon Theme from" << dir.path();
+    }
+
+    themeDebugLog() << "<> Icon themes initialized.";
+}
+
+void ThemeManager::initializeWidgets()
+{
+    themeDebugLog() << "<> Initializing Widget Themes";
+    themeDebugLog() << "Loading Built-in Theme:" << addTheme(std::make_unique<SystemTheme>());
+    auto darkThemeId = addTheme(std::make_unique<DarkTheme>());
+    themeDebugLog() << "Loading Built-in Theme:" << darkThemeId;
+    themeDebugLog() << "Loading Built-in Theme:" << addTheme(std::make_unique<BrightTheme>());
+
+    // TODO: need some way to differentiate same name themes in different subdirectories (maybe smaller grey text next to theme name in
+    // dropdown?)
+
+    if (!m_applicationThemeFolder.mkpath("."))
+        themeWarningLog() << "Couldn't create theme folder";
+    themeDebugLog() << "Theme Folder Path: " << m_applicationThemeFolder.absolutePath();
+
+    QDirIterator directoryIterator(m_applicationThemeFolder.path(), QDir::Dirs | QDir::NoDotAndDotDot, QDirIterator::Subdirectories);
```

Comment:
Can you please remove the `QDirIterator::Subdirectories` ?
I removed that in the catpacks PR (as a request from Scrumplex:https://github.com/PrismLauncher/PrismLauncher/pull/1253#discussion_r1267803706)

Reply:
Oh, I don't know what happened there as I didn't specifically add it here